hack/test/e2e.sh

```diff
@@ -0,0 +1,131 @@
+e2e_run "kubectl apply -f /e2emanifests/provider-components.yaml -f /e2emanifests/capi-secrets.yaml"
+
+##Wait for talosconfig in cm then dump it out
+e2e_run "timeout=\$((\$(date +%s) + ${TIMEOUT}))
+		 until kubectl wait --timeout=1s --for=condition=Ready -n cluster-api-provider-talos-system pod/cluster-api-provider-talos-controller-manager-0
+		 do
+		   if  [[ \$(date +%s) -gt \$timeout ]] 
+		   then
+		     exit 1
+		   fi
+
+		   kubectl describe po -n cluster-api-provider-talos-system cluster-api-provider-talos-controller-manager-0
+		   sleep 5
+		 done"
+
+##Wait for cluster-api-provider-talos-controller-manager-0 to be ready
+e2e_run "kubectl wait --timeout=${TIMEOUT}s --for=condition=Ready -n cluster-api-provider-talos-system pod/cluster-api-provider-talos-controller-manager-0"
+
+##Create cluster and create machines in packet
+##TODO: Accept list of IPs as env var for the master-ips bit.
+git clone --branch v${CAPT_VERSION} https://github.com/talos-systems/cluster-api-provider-talos.git ${TMP}/cluster-api-provider-talos
+sed -i "s/\[x.x.x.x, y.y.y.y, z.z.z.z\]/\[139.178.69.76\]/" ${TMP}/cluster-api-provider-talos/config/samples/cluster-deployment/packet/master-ips.yaml
+sed -i "s/{{PROJECT_ID}}/${PACKET_PROJECT_ID}/g; s/{{PXE_SERVER}}/${PACKET_PXE_SERVER}/g;" ${TMP}/cluster-api-provider-talos/config/samples/cluster-deployment/packet/platform-config-*.yaml
+
+##Download kustomize and template out capi cluster, then deploy it
+e2e_run "apt update && apt install wget
+		 wget -O /usr/local/bin/kustomize https://github.com/kubernetes-sigs/kustomize/releases/download/v${KUSTOMIZE_VERSION}/kustomize_${KUSTOMIZE_VERSION}_linux_amd64
+  	     chmod +x /usr/local/bin/kustomize
+         kustomize build ${TMP}/cluster-api-provider-talos/config/samples/cluster-deployment/packet > /e2emanifests/packet-cluster.yaml
+		 kubectl apply -f /e2emanifests/packet-cluster.yaml"
+
+##Wait for talosconfig in cm then dump it out
+e2e_run "timeout=\$((\$(date +%s) + ${TIMEOUT}))
+		 until kubectl get cm -n cluster-api-provider-talos-system talos-test-cluster-master-0
+		 do
+		   if  [[ \$(date +%s) -gt \$timeout ]] 
+		   then
+		     exit 1
+		   fi
+		   sleep 5
+		 done"
+
+e2e_run "kubectl get cm -n cluster-api-provider-talos-system talos-test-cluster-master-0 -o jsonpath='{.data.talosconfig}' > ${TALOSCONFIG}-capi"
+
+##Wait for kubeconfig from capi master-0
+e2e_run "timeout=\$((\$(date +%s) + ${TIMEOUT}))
+         until /bin/osctl --talosconfig ${TALOSCONFIG}-capi kubeconfig > ${KUBECONFIG}-capi
+         do 
+		   if  [[ \$(date +%s) -gt \$timeout ]]
+	       then
+	         exit 1
+	       fi
+           sleep 5
+         done"
+
+# ## Wait for 2 nodes to check in
+e2e_run "timeout=\$((\$(date +%s) + ${TIMEOUT}))
+         until KUBECONFIG=${KUBECONFIG}-capi kubectl get nodes -o json | jq '.items | length' | grep ${NUM_NODES} >/dev/null
+         do 
+		   if  [[ \$(date +%s) -gt \$timeout ]]
```

Comment:
nit: Align `if` with the `then`.

Reply:
Should be good now. Was just some tabs vs. space nonsense in vscode